src/applications/personalization/preferences/containers/PreferencesWidget.jsx

```diff
@@ -48,50 +61,108 @@ class PreferencesWidget extends React.Component {
     // Create new message timer
     const savedMessageTimer = setTimeout(
       () => this.setState({ savedMessage: false }),
-      5000,
+      ALERT_DELAY,
     );
     // Set new message timer to state
     this.setState({ savedMessageTimer });
   };
 
-  handleRemove = async slug => {
-    await this.props.setPreference(slug, false);
+  handleRemove = async code => {
+    await this.props.setPreference(code, false);
     this.props.savePreferences(this.props.preferences.dashboard);
     this.setSavedMessage();
   };
 
-  handleViewToggle = slug => {
+  handleViewToggle = code => {
     this.setState({
-      [slug]: !this.state[slug],
+      [code]: !this.state[code],
     });
   };
 
-  render() {
+  handleCloseAlert = () => {
+    clearTimeout(this.state.savedMessageTimer);
+    this.setState({ savedMessage: false });
+  };
+
+  renderContent() {
     const {
-      preferences: { dashboard },
+      preferences: {
+        dashboard,
+        userBenefitsLoadingStatus: loadingStatus,
+        saveStatus,
+      },
     } = this.props;
-    const { savedMessage } = this.state;
-    // do not show in production
-    if (environment.isProduction()) {
-      return null;
-    }
     const selectedBenefits = benefitChoices.filter(
-      item => !!dashboard[item.slug],
+      item => !!dashboard[item.code],
     );
     const hasSelectedBenefits = !!selectedBenefits.length;
     let selectedBenefitAlerts = selectedBenefits
       .filter(item => !!item.alert)
       .map(item => item.alert);
     selectedBenefitAlerts = deduplicate(selectedBenefitAlerts);
+
+    if (loadingStatus === LOADING_STATES.pending) {
```

Comment:
Might be more declarative to use ternaries in a single return block e.g. 
`{saveState === LOADING_STATES.error && <Component>}`

even better would be to move this logic into the `mapStateToProps` or the reducer so that these states can be named e.g. 
`showSaveError` could be a prop and the reducer could create that by checking the save status.

Reply:
You're probably right and this isn't the most idiomatic React, but it's so easy for my brain the reason about :)